### PR TITLE
fixes #1244 - add smart proxy feature refresh link + API call

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

File: app/controllers/smart_proxies_controller.rb [hunk 1]
```diff
@@ -22,7 +22,15 @@ def edit
...
   def ping
     respond_to do |format|
-      format.json {render :json => errors_hash(@proxy.ping)}
+      format.json {render :json => errors_hash(@proxy.refresh)}
+    end
+  end
+
+  def refresh
+    if @proxy.refresh.blank? && @proxy.save
+      process_success :object => @proxy, :success_msg => _("Successfully refreshed features from %s.") % @proxy.name
```
Comment (skottler) [inline app/controllers/smart_proxies_controller.rb]: It'd be ideal if there was a different message for when the features change vs. when they don't.

**Comment by skottler** [comment]
Overall looks good + works well. Just that one comment about the message on refresh and I think it's ready for merge.

File: app/controllers/smart_proxies_controller.rb (hunk 1 above)
Comment (domcleal) [inline app/controllers/smart_proxies_controller.rb]: @skottler good idea, updated :)

**Comment by skottler** [comment]
Much better. Thanks @domcleal!

Ready for merge once the build goes green.

**Comment by skottler** [comment]
Merged as c3d8dd61d39ddba9d6d0698b8eda245de23453c8. Thanks!

**Pull request closed without merging**